File: Sources/NSMatrix-Extension.m
#import "NSMatrix-Extension.h"

@implementation NSMatrix (Extension)
- (void)setCellColor:(NSColor*)color
{
    [(NSArray<NSButtonCell*>*)(self.cells) enumerateObjectsUsingBlock:^(NSButtonCell * _Nonnull cell, NSUInteger idx, BOOL * _Nonnull stop) {
        NSMutableAttributedString *title =
        [[NSMutableAttributedString alloc] initWithAttributedString:cell.attributedTitle];
        
        [title addAttribute:NSForegroundColorAttributeName
                      value:color
                      range:NSMakeRange(0, title.length)];
        
        cell.attributedTitle = title;
    }];
}

- (void)setCellFont:(NSFont*)font
{
    [(NSArray<NSButtonCell*>*)(self.cells) enumerateObjectsUsingBlock:^(NSButtonCell * _Nonnull cell, NSUInteger idx, BOOL * _Nonnull stop) {
        NSMutableAttributedString *title =
        [[NSMutableAttributedString alloc] initWithAttributedString:cell.attributedTitle];
        
        [title addAttribute:NSFontAttributeName
                      value:font
                      range:NSMakeRange(0, title.length)];
        
        cell.attributedTitle = title;
    }];
}

@end

File: Sources/Extensions/AppKit/NSMatrix-Extension.swift
import Cocoa

extension NSMatrix {
    @objc func setCellColor(_ color: NSColor) {
        for case let cell as NSButtonCell in self.cells {
            let title = NSMutableAttributedString(attributedString: cell.attributedTitle)

            title.addAttribute(
                .foregroundColor,
                value: color,
                range: NSRange(location: 0, length: title.length))

            cell.attributedTitle = title
        }
    }

    @objc func setCellFont(_ font: NSFont) {
        for case let cell as NSButtonCell in self.cells {
            let title = NSMutableAttributedString(attributedString: cell.attributedTitle)

            title.addAttribute(
                .font,
                value: font,
                range: NSRange(location: 0, length: title.length))

            cell.attributedTitle = title
        }
    }
}
